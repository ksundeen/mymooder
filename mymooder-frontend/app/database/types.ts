import {Path as RePath} from 'react-native-redash';


export type MoodValue = {
    id: number;
    latitude_x: number,
    longitude_y: number,
    name: string,
    datetime: string,
    calmness_score: number,
    happy_score: number,
    people: string,
    activities: string,
    personal_weather_rating: string,
    api_weather_rating: string,
    api_weather_temperature: number,
    notes: string
};

export type MoodValueNumber = {
    calmness_score: number,
    happy_score: number,
    api_weather_temperature: number,
};

export type MoodLabel = {
    label: string, 
    value: string
};

export type DateValues = Date;

export type HappyValues = number;

export type CalmValues = number;

export type PeopleValues = string;

export type ActivitiesValues = string;

export type WeatherValues = string;

export type LocationValues = {
    latitude: number,
    longitude: number
};
export type WeatherAPIValues = {
    weatherAPIValues: string,
    weatherAPITemp: number
};
export type DateRange = {
    maxIsoDate: string, 
    minIsoDate: string
};
export type HappyRange = {
    maxVal: number,
    minVal: number,
};
export type CalmRange = {
    maxVal: number,
    minVal: number,
};

export type PeopleSearch = string[];

export type ActivitiesSearch = string[];

export type PersonalWeatherSearch = string[];

export type WeatherApiSearch = {
    weatherRating: string[],
    tempRange: {
        maxTemp: number,
        minTemp: number
    }
};

export type ChartPointNumber = {
    date: string,
    value: number,
    label: string, 
    labelTextStyle: {color: string, width: number}
};

export type GraphData = {
    max: number,
    min: number,
    curve: RePath,
};

export type ChartPointCategory = {
    date: string,
    value: string
}

export type NumberChart = {
    name: string, 
    data: ChartPointNumber[], 
    maxDate: Date, 
    minDate: Date, 
    maxVal: number, 
    minVal: number
} 
