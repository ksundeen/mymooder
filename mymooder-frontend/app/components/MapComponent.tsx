import { 
  LatLng, 
  LeafletView, 
  MapMarker, 
  MapShape, 
  WebViewLeafletEvents, 
  WebviewLeafletMessage } from 'react-native-leaflet-view-2';
// import { favoritePlaceData } from '@/assets/data/favorite-places';
import { Dimensions, StyleSheet, View } from 'react-native';
import { useState } from 'react';
import Svg from 'react-native-svg';
import { LocationValues, MoodValue } from '../database/interfaces/interfaces';
import ModalInfoBox from './modals/ModalInfoBox';
// import { IFrameWebView } from './IFrameWebView';

const {height, width} = Dimensions.get("window");              

export function MapComponent({setLocationsFromMapCaller, 
                              mapData, 
                              clusterIconsVisible, 
                              mapCenter, 
                              mapShapes, 
                              mapMarkers}: 
  { setLocationsFromMapCaller: Function, 
    mapData: MoodValue[], 
    clusterIconsVisible: boolean, 
    mapCenter: LatLng | null, 
    mapShapes: MapShape[], 
    mapMarkers: MapMarker[]}
  ) {
  
  // To send locations clicked on map to Mood Screen
  const [selectedMoodLocation, setSelectedMoodLocation] = useState<MoodValue | null>(null);
  const [selectedLocationValues, setSelectedLocationValues] = useState<LocationValues | null>(null);
  
  // Whether to show the clicked icon info box modal; state passed to modal component
  const [showInfoBoxModal, setShowInfoBoxModal] = useState<boolean>(false);

  // const [mapPeriod, setMapPeriod] = React.useState<Period>(Period.week);

  // useEffect(() => {
  //   const fetchData = async () => {
  //     if (chartPeriod === Period.week) {
  //       const { startDate, endDate } = getWeekRange(currentDate);
  //       setCurrentEndDate(() => new Date(startDate));
  //       const data = await fetchWeeklyData(startDate, endDate, transactionType);
  //       console.log("Data before process", data);
  //       setBarData(processWeeklyData(data, transactionType));
  //       setChartKey((prev) => prev + 1);
  //     }
  //   };
  //   fetchData();
  // }, [chartPeriod, currentDate, transactionType]);

  const onMessageReceived = (message: WebviewLeafletMessage) => {
    switch (message.event) {
      case WebViewLeafletEvents.ON_MAP_MARKER_CLICKED:
        const idMarkerClicked: string = message.payload?.mapMarkerID as string;
        const selectedPoint: MoodValue = mapData.filter(feature => feature.id === Number(idMarkerClicked))[0]
        setSelectedLocationValues(null)
        setSelectedMoodLocation(selectedPoint)
        setShowInfoBoxModal(true)
          // Alert.alert(`Map Touched at:`, `${position.lat}, ${position.lng} for id: ${id}`);

        break;
      case WebViewLeafletEvents.ON_MAP_TOUCHED:
        const position: LatLng = message.payload?.touchLatLng as LatLng;
        console.log(JSON.stringify(position))
          setSelectedMoodLocation(null)
          console.log(JSON.stringify(position))
          setSelectedLocationValues({latitude: Number(position.lat), longitude: Number(position.lng)})
          setShowInfoBoxModal(true)
        break;
      // case WebViewLeafletEvents.ON_ZOOM_END:
      //   const zoomLevel: number = message.payload
      //     ?.zoom as number;
      //     // setZoomLevel(zoomLevel);
      //     // Alert.alert(`New zoom leve:`, `${zoomLevel}`);

      //   break;
      default:
        console.log("App received", message);
    }
  };

    // const mapLayers = [
    //   {
    //     baseLayerName: 'Openstreet Map',  // the name of the layer, this will be seen in the layer selection control
    //     baseLayerIsChecked: 'false',  // if the layer is selected in the layer selection control
    //     layerType: 'TileLayer',  // Optional: a MapLayerType enum specifying the type of layer see "Types of Layers" below. Defaults to TILE_LAYER
    //     baseLayer: true,
    //     // url of tiles
    //     url: 'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
    //     // attribution string to be shown for this layer
    //     attribution:
    //       '&amp;copy <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors'
    //   }
    //   ,{
    //   baseLayerName: 'Mapbox Aerial',  // the name of the layer, this will be seen in the layer selection control
    //   baseLayerIsChecked: 'false',  // if the layer is selected in the layer selection control
    //   layerType: 'TileLayer',  // Optional: a MapLayerType enum specifying the type of layer see "Types of Layers" below. Defaults to TILE_LAYER
    //   baseLayer: true,
    //   // url of tiles
    //   url: `https://api.tiles.mapbox.com/v4/mapbox.streets/{z}/{x}/{y}.png?access_token=${process.env.MAPBOX_TOKEN}`,
    //   attribution: ''
    // }
  // ]

  return (
    <View style={styles.container}>
      <Svg width={width} height={height} />
      {/* <IFrameWebView 
          source={ */}
        {/* <SegmentedControl
          values={["Week", "Month", "Year"]}
          style={{ marginBottom: 16 }}
          selectedIndex={currentTab}
          onChange={(event) => {
            const index = event.nativeEvent.selectedSegmentIndex;
            if (index === 0) {
              setChartPeriod(Period.week);
              } else if (index === 1) {
                setChartPeriod(Period.month);
                } else {
                  setChartPeriod(Period.year);
              }
              }}
              /> */}
            <ModalInfoBox
              locationValues={selectedLocationValues}
              moodValue={selectedMoodLocation}
              showInfoBoxModal={showInfoBoxModal}
              setShowInfoBoxModalCaller={setShowInfoBoxModal}
              setLocationsFromMapToMoodCaller={setLocationsFromMapCaller}            
            />
            {mapCenter ? 
              <LeafletView
                onMessageReceived={onMessageReceived}
                // doDebug={true}
                mapMarkers={clusterIconsVisible ? mapMarkers : []}
                mapShapes={clusterIconsVisible ? [] : mapShapes}
                // renderLoading={loader}
                // mapLayers={mapLayers}
                mapCenterPosition={mapCenter}
                zoomControl={true}
                zoom={7}
            ></LeafletView>
            :
            <LeafletView
              onMessageReceived={onMessageReceived}
              // doDebug={true}
              mapMarkers={clusterIconsVisible ? mapMarkers : []}
              mapShapes={clusterIconsVisible ? [] : mapShapes}
              // renderLoading={loader}
              // mapLayers={mapLayers}
              zoomControl={true}
              zoom={7}
            ></LeafletView>
            }
        {/* }

       >
       </IFrameWebView>  */}
    </View>
  );
};

export default MapComponent;


const styles = StyleSheet.create({
  container: {
     flex: 1,
     alignItems: 'center',
     justifyContent: 'center',
  },
  legend:{
    height: 300,
    width: 375,
  },
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
  },
});