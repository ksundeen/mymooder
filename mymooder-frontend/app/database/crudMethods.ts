import * as SQLite from "expo-sqlite";
import { useState } from "react";
import { MoodValue } from "./interfaces/interfaces";

export function crudMoodValuesMethods() {
    // const [moodValues, setMoodValues] = useState<MoodValue[]>([]);
    
    /**
     * Whenever the mood_values table has mutated, we need to fetch the data set again order to sync DB -> UI State
    */
   // const fetchMoodValues = (txn: SQLite.SQLiteExecuteAsyncResult) => {
    //     txn.execAsync("SELECT * FROM mood_values;") => {
        //         // TODO - I'm not sure how to set the data once received. 
        //         setMoodValues(_array)
        //     }
        // };
        
        
    // const getMoodValues = async (db: SQLite.SQLiteDatabase) => {
    //     let moodValues: MoodValue[] = [];
    //     const allRows: MoodValue[] = await db.getAllAsync("SELECT * FROM mood_values;");
    //     // setMoodValues(allRows);
    //     // moodValues = allRows
    // }
    const getMoodValues = async (db: SQLite.SQLiteDatabase) => {    
        const result: MoodValue[] = await db.getAllAsync<MoodValue>(
          `SELECT * FROM mood_values WHERE id > ?;`, [0])
          return result
        };

    const addMoodValue = (db: SQLite.SQLiteDatabase, moodValue: MoodValue) => {
        db.withExclusiveTransactionAsync(async (txn) => {
            await txn.runAsync(`INSERT INTO mood_values (name, latitude_x, longitude_y, datetime, calmness_score, happy_score, people, activities, personal_weather_rating, api_weather_rating, api_weather_temperature, notes)  
                VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);`,
                moodValue.name,
                moodValue.latitude_x, 
                moodValue.longitude_y, 
                moodValue.datetime,
                moodValue.calmness_score, 
                moodValue.happy_score, 
                moodValue.people, 
                moodValue.activities, 
                moodValue.personal_weather_rating, 
                moodValue.api_weather_rating, 
                moodValue.api_weather_temperature, 
                moodValue.notes
            ).then(response => console.log(response))
            .catch(error => console.log(error));
        });
    };

    const updateMoodValue = (db: SQLite.SQLiteDatabase, moodValue: MoodValue) => {
        db.withExclusiveTransactionAsync(async (txn) => {
            await txn.runAsync(`UPDATE mood_values 
                    SET name = ?,
                    latitude_x = ?,
                    longitude_y = ?,
                    datetime = ?,
                    calmness_score = ?,
                    happy_score = ?,
                    people = ?,
                    activities = ?,
                    personal_weather_rating = ?,
                    api_weather_rating = ?,
                    api_weather_temperature = ?,
                    notes = ?               
                WHERE id = ?;`, 
                moodValue.name, 
                moodValue.latitude_x,
                moodValue.longitude_y, 
                moodValue.datetime,
                moodValue.calmness_score, 
                moodValue.happy_score,
                moodValue.people,
                moodValue.activities,
                moodValue.personal_weather_rating,
                moodValue.api_weather_rating,
                moodValue.api_weather_temperature,
                moodValue.notes
                );
        });
    };

  const deleteMoodValue = (db: SQLite.SQLiteDatabase, id: number) => {
        db.withExclusiveTransactionAsync(async (txn) => {
            await txn.runAsync('DELETE FROM mood_values WHERE id = ?', id);
        });
  };

  return {
    getMoodValues,
    addMoodValue,
    updateMoodValue,
    deleteMoodValue,
  };
}