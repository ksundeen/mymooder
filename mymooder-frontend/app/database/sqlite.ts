import * as SQLite from "expo-sqlite";
import { favoritePlaceData } from '@/assets/data/favorite-places';
import { useEffect, useState } from "react";


// Connection is initialised globally
// const db = await SQLite.openDatabaseAsync("MoodValues");

// export async function db() {
//     return await SQLite.openDatabaseAsync("MoodValues");
// };

/**
 * If you have a existing database this is where you would import it,
 * otherwise this is where you would create tables and seed DB.
 */
export const initDatabase = (db: SQLite.SQLiteDatabase) => {
    let seedSqlMoodValues: string = '';
    favoritePlaceData.features.forEach(feature => {
        let geom = feature.geometry;
        let props = feature.properties;
        seedSqlMoodValues += `INSERT INTO mood_values (name, latitude_x, longitude_y, datetime, calmness_score, happy_score, people, activities, personal_weather_rating, api_weather_rating, api_weather_temperature, notes) VALUES 
                 ('${props.name}', 
                ${geom.coordinates[1]}, 
                ${geom.coordinates[0]}, 
                '${props.datetime}',
                ${props.calmness_score}, 
                ${props.happy_score}, 
                '${props.people}', 
                '${props.activities}', 
                '${props.personal_weather_rating}', 
                '${props.api_weather_rating}', 
                ${props.api_weather_temperature}, 
                '${props.notes}'); `
    });
    console.log(seedSqlMoodValues);

    db.withExclusiveTransactionAsync(async () => {
        await db.execAsync(
        `DROP TABLE IF EXISTS mood_values; CREATE TABLE IF NOT EXISTS mood_values 
            (
                id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, 
                name TEXT,
                latitude_x REAL,
                longitude_y REAL,
                datetime TEXT, 
                calmness_score INTEGER, 
                happy_score INTEGER, 
                people TEXT, 
                activities TEXT, 
                personal_weather_rating TEXT, 
                api_weather_rating TEXT, 
                api_weather_temperature INTEGER, 
                notes TEXT
            ); ${seedSqlMoodValues}`
        );
        console.log('LOADED SEED DATA');
    })
        .then(() => true)
        .catch((e) => console.error(e));
};

export default initDatabase;