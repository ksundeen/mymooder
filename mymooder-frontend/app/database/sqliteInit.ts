import * as SQLite from "expo-sqlite";

import { favoritePlaceData } from '@/assets/data/favorite-places';

/**
 * If you have a existing database this is where you would import it,
 * otherwise this is where you would create tables and seed DB.
 */
export const initDatabaseIfNeeded = async (db: SQLite.SQLiteDatabase) => {
    let seedSqlMoodValues: string = '';
    favoritePlaceData.features.forEach(feature => {
        let geom = feature.geometry;
        let props = feature.properties;
        seedSqlMoodValues += ` 
        INSERT INTO mood_values (name, latitude_x, longitude_y, datetime, calmness_score, happy_score, people, activities, personal_weather_rating, api_weather_rating, api_weather_temperature, notes) 
        VALUES 
        ('${props.name}', ${geom.coordinates[1]}, ${geom.coordinates[0]}, '${props.datetime}', ${props.calmness_score}, ${props.happy_score}, '${props.people}', '${props.activities}', '${props.personal_weather_rating}', '${props.api_weather_rating}', ${props.api_weather_temperature}, '${props.notes}'); `
    });
    console.log(seedSqlMoodValues);

    const DATABASE_VERSION = 1;

    interface dbVersionResult {
        // user_version: number,
        currentDbVersion: number
    }

    const getDbVersion = async () => {
        // await db.getFirstAsync<{ user_version: number }>(
        return await db.getFirstAsync<{ currentDbVersion: number }>(
            'PRAGMA user_version'
        );
    }
    const dbVersion: dbVersionResult = await getDbVersion();

    if (dbVersion) {
        console.log("dbVersion: ", dbVersion)
        
        if (dbVersion.currentDbVersion >= DATABASE_VERSION) {
        return;
        }
        if (dbVersion.currentDbVersion === 0) {

            await db.runAsync(
                `PRAGMA journal_mode = 'wal';
                DROP TABLE IF EXISTS mood_values; 
                CREATE TABLE IF NOT EXISTS mood_values 
                    (
                        id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, 
                        name TEXT,
                        latitude_x REAL,
                        longitude_y REAL,
                        datetime TEXT, 
                        calmness_score INTEGER, 
                        happy_score INTEGER, 
                        people TEXT, 
                        activities TEXT, 
                        personal_weather_rating TEXT, 
                        api_weather_rating TEXT, 
                        api_weather_temperature INTEGER, 
                        notes TEXT
                    ); ? `, seedSqlMoodValues
                );

            console.log('LOADED SEED DATA');
            dbVersion.currentDbVersion = 1;
            await db.execAsync(`PRAGMA user_version = ${DATABASE_VERSION}`);

            const dbVersionFinal: dbVersionResult = await getDbVersion();
            console.log("dbVersionFinal: ", dbVersionFinal)
        }
    }
};

export const loadData = async (db: SQLite.SQLiteDatabase) => {
    let seedSqlMoodValues: string = '';
    favoritePlaceData.features.forEach(feature => {
        let geom = feature.geometry;
        let props = feature.properties;
        seedSqlMoodValues += ` 
        INSERT INTO mood_values (name, latitude_x, longitude_y, datetime, calmness_score, happy_score, people, activities, personal_weather_rating, api_weather_rating, api_weather_temperature, notes) 
        VALUES 
        ('${props.name}', ${geom.coordinates[1]}, ${geom.coordinates[0]}, '${props.datetime}', ${props.calmness_score}, ${props.happy_score}, '${props.people}', '${props.activities}', '${props.personal_weather_rating}', '${props.api_weather_rating}', ${props.api_weather_temperature}, '${props.notes}'); `
    });
    console.log(seedSqlMoodValues);

    await db.runAsync(
        `DROP TABLE IF EXISTS mood_values; 
        CREATE TABLE IF NOT EXISTS mood_values 
            (
                id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, 
                name TEXT,
                latitude_x REAL,
                longitude_y REAL,
                datetime TEXT, 
                calmness_score INTEGER, 
                happy_score INTEGER, 
                people TEXT, 
                activities TEXT, 
                personal_weather_rating TEXT, 
                api_weather_rating TEXT, 
                api_weather_temperature INTEGER, 
                notes TEXT
            ); ? `, seedSqlMoodValues
        );

    console.log('LOADED SEED DATA');
};

export default initDatabaseIfNeeded;