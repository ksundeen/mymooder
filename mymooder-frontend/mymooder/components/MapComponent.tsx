import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { LatLng, LeafletView } from 'react-native-leaflet-view';

const DEFAULT_COORDINATE: LatLng = {
  lat: 43.233224,
  lng: -89.346395,
};

export function MapComponent(dimensions: any) {
  return (
    <View>
        <LeafletView
            mapMarkers={[
                {
                    position: DEFAULT_COORDINATE,
                    icon: '📍',
                    size: [32, 32],
                },
            ]}
            mapCenterPosition={DEFAULT_COORDINATE}
        />
    </View>
  );
}

const styles = StyleSheet.create({});

export default MapComponent;