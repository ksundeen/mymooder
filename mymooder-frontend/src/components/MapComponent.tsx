import * as React from 'react';
import { LatLng, LeafletView, MapMarker } from 'react-native-leaflet-view-2';
import favoritePlaces from '@/assets/data/favorite-places';
import { Colors } from '@/constants/Colors';

export function MapComponent() {
  const DEFAULT_COORDINATE: LatLng = {
    lat: 45.04105544754055,
    lng: -90.8013689517975, 
  };

  let mapMarkers: MapMarker[] = []

  for (let i = 0; i < favoritePlaces.features.length; i++) {
    mapMarkers.push({
      position: {
        lng: favoritePlaces.features[i].geometry.coordinates[0], 
        lat: favoritePlaces.features[i].geometry.coordinates[1]
      },
      icon: '📍',
      size: [32, 32],
      color: Colors.lightBlue
    })
  }

  return (
      <LeafletView
          mapMarkers={mapMarkers}
          mapCenterPosition={DEFAULT_COORDINATE}
          zoomControl={true}
          zoom={7}
      />
  );
}

export default MapComponent;
