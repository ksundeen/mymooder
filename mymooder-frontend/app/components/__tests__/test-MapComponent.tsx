import * as React from 'react';
import { render } from '@testing-library/react-native'

// Mocks the WebView component, which fails with "'RNCWebView' could not be found"
jest.mock('react-native-webview', () => {
  const MockWebView = jest.requireActual('react-native').View;

  return {
    __esModule: true,
    WebView: MockWebView,
    default: MockWebView,
  };
});

import { MapComponent } from '../MapComponent';
import { MoodValue } from '@/app/database/types';
import { MapMarker, MapShape } from 'react-native-leaflet-view-2';

it(`renders correctly`, () => {
  const mapData: MoodValue[] = [
    {
      id: 1,
      latitude_x: 45.5,
      longitude_y: -84.5,
      name: 'Test',
      datetime: '01-04-2024',
      calmness_score: 3,
      happy_score: 5,
      people: 'me',
      activities: 'running',
      personal_weather_rating: 'sunny',
      api_weather_rating: 'parly sunny',
      api_weather_temperature: 75,
      notes: ''
    }
  ]

  const setLocationsFromMapToMoodCaller = () => { console.log('called setLocationsFromMapToMoodCaller()')}
  const clusterIconsVisible=false
  const mapCenter = [45.0, -92.0]
  const mapShapes: MapShape[] = []
  const mapMarkers: MapMarker[] = []

  const tree = render(
    <MapComponent 
      mapData={mapData}
      setLocationsFromMapToMoodCaller={setLocationsFromMapToMoodCaller}
      clusterIconsVisible={clusterIconsVisible}
      mapCenter={mapCenter}
      mapShapes={mapShapes}
      mapMarkers={mapMarkers}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
