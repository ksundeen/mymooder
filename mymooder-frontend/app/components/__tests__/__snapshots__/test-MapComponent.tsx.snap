// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RNSVGSvgView
    bbHeight={1334}
    bbWidth={750}
    focusable={false}
    height={1334}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
    width={750}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGSvgView>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
  <View
    allowFileAccess={true}
    allowFileAccessFromFileURLs={true}
    allowUniversalAccessFromFileURLs={true}
    containerStyle={
      {
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    domStorageEnabled={true}
    javaScriptEnabled={true}
    onMessage={[Function]}
    originWhitelist={
      [
        "*",
      ]
    }
    renderLoading={[Function]}
    source={1}
    startInLoadingState={true}
  />
</View>
`;
